src/rsa/padding.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2015-2016 Brian Smith.
+//
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND AND THE AUTHORS DISCLAIM ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY
+// SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
+// OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+// CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+use untrusted;
+use digest;
+
+#[cfg(feature = "rsa_signing")]
+use rand;
+
+#[cfg(feature = "rsa_signing")]
+pub trait Encoding {
+    fn encode(&self, msg: &[u8], out: &mut [u8], rng: &rand::SecureRandom)
+              -> Result<(), ()>;
+}
+
+pub trait Verification {
+    fn verify(&self, msg: untrusted::Input, encoded: untrusted::Input) 
```

Comment:
Okay, maybe that makes sense for PSS. Still, this doesn't seem as good a description of the PKCS1 padding operations, which also need to be represented, right?

At a high level, we're fitting a hash with variable size into a larger-sized block (by key size). To me, "encoding" invokes the idea of something that keeps the size the same, or has a linear relation between input and output size.

In the end, I guess this isn't public API anyway, so the high-level/end-user view is maybe less important than following the spec/using terminology crypto-engineers use.